packages/grid/_modules_/grid/components/containers/GridFooterContainer.tsx

```diff
@@ -7,6 +7,6 @@ export const GridFooterContainer = React.forwardRef<HTMLDivElement, GridFooterCo
   function GridFooterContainer(props: GridFooterContainerProps, ref) {
     const { className, ...other } = props;
 
-    return <div ref={ref} className={clsx('MuiDataGrid-footer', className)} {...other} />;
+    return <div ref={ref} className={clsx('MuiGridFooterContainer', className)} {...other} />;
```

Comment:
These were meant to be reusable components. That is actually not part of what we discussed, I just had an idea and put it into a discussion. I can revert those changes (for these container components) and just leave the rest of the classes.

If on the other hand, we can solve the problem now that would be great. I'm more in favour for `MuiGridFooterContainer-root`

Reply:
Actually, it's the same with `.MuiGridToolbarContainer`, this is not matching any of the two patterns.

I would personally vote in favor of standalone component every time the component is likely to be imported by the developers (e.g. documented in the docs). If it has its own styles, we don't even have the choice, it should be standalone. 

Here, it seems to be part of the overall structure of the main data grid component, so I would vote for sub-component even if it can be composed on its own.

cc @mui-org/x for more feedback.

Reply:
Yeah, if it's a standalone component I think we should strictly follow the same naming from the core. In this context, using `.MuiGridFooterContainer-root` and not BEM. My concern is that we'll end up having two conventions in the grid package.

Reply:
@m4theushw note that this double convention problem is also present on the date picker, 

There is an extra concern here. The style of the component are hosted by the root. Render it outside and it will broke.

Reply:
So do we agree to call those container components like `.MuiGridFooterContainer-root` for example?

Reply:
I vote against `.MuiGridFooterContainer-root`/ In favor of `.MuiDataGrid-footerContainer` because the CSS of this class name is own by GridRootStyles.

Reply:
Renamed